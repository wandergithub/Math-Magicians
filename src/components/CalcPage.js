import Calculator from './Calculator';
import Header from './Header';

const CalcPage = () => (
  <div>
    <Header />
    <Calculator />
  </div>
);

export default CalcPage;
