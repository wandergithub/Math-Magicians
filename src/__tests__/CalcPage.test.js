import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CalcPage from '../components/CalcPage';

describe("Calculator Page tests", () => {
  it("Renders correctly", () => {
    const tree = renderer 
      .create(
        <MemoryRouter>
          <CalcPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
