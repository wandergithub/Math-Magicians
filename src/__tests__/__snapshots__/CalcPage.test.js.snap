// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Page tests Renders correctly 1`] = `
<div>
  <header
    className="nav-bar"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="link-container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="mid-b"
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </header>
  <div
    className="calculator-container"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      className="calculatorContainer"
    >
      <div
        className="outputContainer"
      >
        <p>
          0
          0
        </p>
      </div>
      <div
        className="buttons-container"
      >
        <div
          className="row"
        >
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="row row-5"
        >
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
